dashboard-improvements/README.md

```diff
@@ -0,0 +1,151 @@
+- [IDs for COPR builds in API endpoint (bug)](https://github.com/packit/packit-service/issues/982)
+  - Does not affect usage of COPR build result page so far (is not used), or is the
+    reason for only one chroot result being displayed?
+
+## Current state
+
+- [Already existing TODO list](https://github.com/packit/dashboard/blob/main/TODO.md)
+- We have empty homepage
+  - I can see a potential in showing some statistics, could be helpful to us
+    and also to users.
+  - We could list recently finished jobs or queue.
+- We have implemented result pages
+  - COPR builds:bug mentioned above
+  - Minimal info for testing farm
+- For COPR builds we link directly COPR logs on the dashboard (recent COPR builds),
+  we should probably address our own result page before the COPR one.
+- Navigation:
+  - From each of those we can get back to the branch or the PR
+  - SRPM results -> {can see logs, download SRPM if available}
+  - COPR results -> {SRPM results, COPR logs, compressed build logs from COPR}
+  - TF results -> dead end
+
+## Removal of the _build result_ webpage
+
+0. Fix COPR build bug mentioned above, so that we can see all chroots.
+1. Set dashboard link for all new commit statuses.
+2. Set 2-week long (conveniently matching sprint) grace period for build result
+   page in service.
+
+   _Argument_: 2 weeks is enough for populating commit statuses with links to dashboard
+   and in majority of cases, old links are mainly used for debugging purposes on
+   our side.
+
+3. Remove build result page from service.
+
+   _Alternatively_: Also redirect all requests from service to dashboard in step 1
+   and keep live for longer period of time.
+
+## Providing more information for dashboard through API
+
+### Allowing read-only access to database
+
+- Suspects: `data` in COPR, Koji and Testing Farm
+- Otherwise there does not appear to be anything sensitive present
+- Potential downsides would be:
+  - Providing information through dashboard that cannot be accessed from API
+  - Additional logic for the dashboard (in case we want to provide information via
+    API too)
+  - Testing
+
+## Authentication against git-forge
+
+- Could be probably generalized to OAuth in regards to expanding to GitLab
+- OAuth
+  - From GitHub we get code (valid for 10 minutes) that can be used to get access
+    token
+  - GitLab appears to work in similar way, there are more options though
+  - For our case we should be fine with one-time codes that would be used client-side
+
+### Possible use-cases
+
+- Retriggering
+
+  We could allow retriggering and be sure that user triggering the build again is
+  authorized to do so (spam/DoS protection)
+
+  Cost on our side:
+  Requires changes in service codebase and also we would need to make sure that
+  retriggering cannot be used outside of the dashboard
+
+  - Auth in API `dashboard <=> service`
+  - we could probably restrict IP address of sender to cluster on service side?
```

Comment:
Require user to be authenticated, disallow retrigger for in-progress jobs - we could use those as roadblocks to avoid disaster.